src/langchain_google_firestore/vectorstores.py

```diff
@@ -0,0 +1,419 @@
+                Defaults to DistanceStrategy.COSINE.
+
+        Raises:
+            ImportError: If the `firestore` package is not found.
+
+        Example:
+            .. code-block:: python
+
+            from langchain_google_firestore.vectorstores import Firestore
+            from langchain.embeddings import GooglePalmEmbeddings
+
+            embeddings = GooglePalmEmbeddings()
+            firestore_vecstore = Firestore(source="my_collection", embeddings=embeddings)
+        """
+        try:
+            from google.cloud import firestore
+        except ModuleNotFoundError as exc:
+            raise ImportError(IMPORT_ERROR_MSG) from exc
+
+        # Check if the client is provided, otherwise create a new client with
+        # the default client info.
+        self.client = client or firestore.Client()
+
+        client_agent = self.client._client_info.user_agent
+        if not client_agent:
+            self.client._client_info.user_agent = USER_AGENT
+        elif USER_AGENT not in client_agent:
+            self.client._client_info.user_agent = " ".join([client_agent, USER_AGENT])
+
+        if isinstance(source, str):
+            self.source = self.client.collection(source)
+        elif isinstance(source, DocumentReference):
+            self.source = source.parent
+        else:
+            self.source = source
+
+        self.embedding = embedding
+        self.content_field = content_field
+        self.metadata_fields = metadata_fields
+        self.ignore_metadata_fields = ignore_metadata_fields
+        self.text_embedding_field = text_embedding_field
+        self.distance_strategy = distance_strategy
+
+    @property
+    def embeddings(self) -> Optional[Embeddings]:
+        return self.embedding
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        collection: Optional[str] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        if isinstance(self.source, CollectionGroup) and not collection:
+            raise ValueError(
+                "The `collection` path must be provided when using CollectionGroup."
+            )
+
+        ids = []
+        db_batch = self.client.batch()
+
+        for batch in more_itertools.chunked(texts, WRITE_BATCH_SIZE):
+            texts_embs = self.embedding.embed_documents(batch)
+            for i, text in enumerate(batch):
+                doc = self.source.document()
+                ids.append(doc.id)
+                data = {
+                    self.content_field: text,
+                    self.text_embedding_field: Vector(texts_embs[i]),
+                }
+                if metadatas:
+                    data.update(metadatas[i])
+                db_batch.set(doc, data)
+            db_batch.commit()
+
+        return ids
+
+    async def aadd_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        collection: Optional[str] = None,
```

Comment:
same comment as above, and also for `kwargs`

Reply:
removing kwargs results in Pylink error:
```
Variadics removed in overriding 'FirestoreVectorStore.add_texts' methodPylint[W0221:arguments-differ](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/arguments-differ.html)
```